p4include/v1model.p4

```diff
@@ -315,84 +371,179 @@ extern Checksum16 {
-@param O          Checksum type; must be bit<X> type.
-@param condition  If 'false' the verification always succeeds.
-@param data       Data whose checksum is verified.
-@param checksum   Expected checksum of the data; note that is must be a left-value.
-@param algo       Algorithm to use for checksum (not all algorithms may be supported).
-                  Must be a compile-time constant.
-*/
+/***
+ * verify_checksum_with_payload is identical in all ways to
+ * verify_checksum, except that it includes the payload of the packet
+ * in the checksum calculation.  The payload is defined as "all bytes
+ * of the packet which were not parsed by the parser".
+ *
+ * Calling verify_checksum_with_payload is only supported in the
+ * VerifyChecksum control.
+ */
 extern void verify_checksum_with_payload<T, O>(in bool condition, in T data, inout O checksum, HashAlgorithm algo);
+
 /**
-Computes the checksum of the supplied data including the payload.
-The payload is defined as "all bytes of the packet which were not parsed by the parser".
-@param T          Must be a tuple type where all the tuple elements are of type bit<W>, int<W>, or varbit<W>.
-                  The total dynamic length of the fields is a multiple of the output size.
-@param O          Output type; must be bit<X> type.
-@param condition  If 'false' the checksum is not changed
-@param data       Data whose checksum is computed.
-@param checksum   Checksum of the data.
-@param algo       Algorithm to use for checksum (not all algorithms may be supported).
-                  Must be a compile-time constant.
-*/
+ * update_checksum_with_payload is identical in all ways to
+ * update_checksum, except that it includes the payload of the packet
+ * in the checksum calculation.  The payload is defined as "all bytes
+ * of the packet which were not parsed by the parser".
+ *
+ * Calling update_checksum_with_payload is only supported in the
+ * ComputeChecksum control.
+ */
 extern void update_checksum_with_payload<T, O>(in bool condition, in T data, inout O checksum, HashAlgorithm algo);
 
+/***
+ * Calling resubmit during execution of the ingress control will,
+ * under certain documented conditions, cause the packet to be
+ * resubmitted, i.e. it will begin processing again with the parser,
+ * with the contents of the packet exactly as they were when it last
+ * began parsing.  The only difference is in the value of the
+ * standard_metadata instance_type field, and any user-defined
+ * metadata fields that the resubmit operation causes to be
+ * preserved.
+ *
+ * Calling resubmit is only supported in the ingress control.  There
+ * is no way to undo its effects once it has been called.  If resubmit
+ * is called multiple times during a single execution of the ingress
+ * control, only one packet is resubmitted, and only the data from the
+ * last such call is preserved.  See the v1model architecture
+ * documentation (Note 1) for more details.
+ */
```

Comment:
If we merge the resubmit patches these will change.

Reply:
Understood.  My goal here was not to get hyper-precise on future possible changes, or the existing broken behavior, but to focus on what resubmit/recirculate/clone do, independent of whether they preserve metadata or not.  I am happy to update the docs when the new solution is decided upon and implemented.